### PR TITLE
feat: support for public inputs columns

### PR DESCRIPTION
**Pull request by Maddiaa0**

### CODE REVIEW

File: pil_analyzer/src/statement_processor.rs [hunk 1]
```diff
@@ -267,10 +267,15 @@ where
         start: usize,
         polynomials: Vec<PolynomialName<T>>,
         polynomial_type: PolynomialType,
+        is_public: bool,
     ) -> Vec<PILItem<T>> {
         polynomials
             .into_iter()
             .flat_map(|PolynomialName { name, array_size }| {
+
+                // TODO: hack: add an is_public modifier to the end of a commited polynomial????
```
Comment (jeanmon) [inline pil_analyzer/src/statement_processor.rs]: Is there a ticket about removing this hack at some point?
On the spike, I did not find any occurence for "__is_public", is this expected?

File: parser/src/powdr.lalrpop [hunk 2]
```diff
@@ -81,9 +81,10 @@ ConstantDefinition: PilStatement<T> = {
 }
 
 PolynomialDefinition: PilStatement<T> = {
-    <@L> PolCol <Identifier> "=" <Expression> => PilStatement::PolynomialDefinition(<>)
+    <loc:@L> PolCol <id:Identifier> "=" <expr:Expression> => PilStatement::PolynomialDefinition(<>)
```
Comment (jeanmon) [inline parser/src/powdr.lalrpop]: is this change really required?

**Review by jeanmon — APPROVED** [review]
LGTM. I am wondering if defining a new PilStatement would not have been the cleanest way but very likely more costly.

File: parser/src/powdr.lalrpop (hunk 2 above)
Comment (Maddiaa0) [reply parser/src/powdr.lalrpop]: Good catch, it was left over from experimentation
